### PR TITLE
Add API for pdf preview

### PR DESCRIPTION
**Pull request by bjosttveit**
## Description

Added a `pdf/preview` API that simply returns the generated PDF as `application/pdf` instead of saving to storage. This is currently disabled in prod just in case, since there is no distinction between the preview and the "real" PDF. And currently the intent is only to be able to get proper PDF preview in dev tools. It also looks for a `frontendVersion` cookie and passes this along if the app is running in LocalTest; it also rewrites `locahost` -> `host.containers.internal` so podman and docker can access a locally running app-frontend.

To check for prod I used `IWebHostEnvironment.IsProduction()`, I have not tested this in prod, but I saw the equivalent `IsDevelopment()` was used elsewhere so I assumed this was the way to do it.

## Related Issue(s)
- #{issue number}

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Altinn.App.Api/Controllers/PdfController.cs [hunk 1]
```diff
@@ -37,27 +38,58 @@ public class PdfController : ControllerBase
...
             _appModel = appModel;
             _dataClient = dataClient;
+            _env = env;
+            _pdfService = pdfService;
+        }
+
+        /// <summary>
+        /// Generate a preview of the PDF for the current task
+        /// </summary>
+        [HttpGet("{org}/{app}/instances/{instanceOwnerPartyId:int}/{instanceGuid:guid}/pdf/preview")]
```
Comment (ivarne) [inline src/Altinn.App.Api/Controllers/PdfController.cs:66]: I have not looked, but there should be some attribute to ensure this endpoint isn't part of the swagger docs.

**Review by ivarne — COMMENTED** [review]
Looks very reasonable

**Comment by ivarne** [comment]
Ja, da gjenstår det "bare" å skrive noen tester. (Jeg kan se på det. Ikke nødvendig at du setter deg inn i dotnet testing om du ikke har veldig lyst)

**Comment by bjosttveit** [comment]
> Ja, da gjenstår det "bare" å skrive noen tester. (Jeg kan se på det. Ikke nødvendig at du setter deg inn i dotnet testing om du ikke har veldig lyst)

Jeg gir det gjerne et forsøk først, har skrevet noen tester før

**Comment by ivarne** [comment]
Kjør på men spør i stedet for å bruke masse tid på å stå fast da.

File: test/Altinn.App.Api.Tests/Controllers/PdfControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        public async Task Request_In_Dev_Should_Generate()
+        {
+            string? requestContent = null;
+            var handler = new Mock<HttpMessageHandler>();
+            handler.Protected().Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
+                .ReturnsAsync(new HttpResponseMessage()
+                {
+                    StatusCode = System.Net.HttpStatusCode.OK,
+                    Content = new StringContent("PDF")
+                })
```
[the remark this answers is not in the document]
Comment (bjosttveit) [reply test/Altinn.App.Api.Tests/Controllers/PdfControllerTests.cs]: Is this relevant in this case?

[the remark this answers is not in the document]
Comment (ivarne) [reply test/Altinn.App.Api.Tests/Controllers/PdfControllerTests.cs]: I don't think so, but I've never been able to get a real answer for what happens when you forget a using statement. I prefer just adding the usings, just to be sure.

**Review by RonnyB71 — APPROVED** [review]
🕹️

**Pull request merged**